R/foreign.R

```diff
@@ -296,7 +296,7 @@ read.graph.ncol <- function(file, predef = character(0), names = TRUE,
   on.exit(.Call(R_igraph_finalizer))
   .Call(
     R_igraph_read_graph_ncol, file, as.character(predef),
-    as.logical(names), as.numeric(weights), as.logical(directed)
+    as.logical(names), weights, as.logical(directed)
```

Comment:
Should this be

```suggestion
    as.logical(names), as.integer(weights), as.logical(directed)
```

?

Reply:
Technically, it doesn't _need_ to be because the constants were explicitly specified as integers (`0L`, `1L`, etc.)  This is what all the auto-generated code looks like as well at the moment.

However, if you want the extra safety of `as.integer`, it doesn't hurt. It's your decision, I don't know what's good R practice. :-)

I'd just like this merged and released.